p2p/src/peers.rs

```diff
@@ -446,13 +446,19 @@ impl Peers {
 		let mut rm = vec![];
 
 		// build a list of peers to be cleaned up
-		for peer in self.connected_peers() {
+		for peer in self.peers.read().unwrap().values() {
```

Comment:
Yeah I see it now - `connected_peers()` is not the set of peers we are connected to, but the set of peers we are connected to that are in state `Connected`...

Reply:
:+1: